flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBWriteBatchWrapper.java

```diff
@@ -113,4 +126,16 @@ public void close() throws RocksDBException {
 		}
 		IOUtils.closeQuietly(batch);
 	}
+
+	private void flushIfNeeded() throws RocksDBException {
+		boolean needFlush = batch.count() == capacity || (batchSize > 0 && batch.getDataSize() >= batchSize);
```

Comment:
No need to check whether `batchSize` is larger than 0 since we already have the `@Nonnegative` annotation.

Reply:
Currently, use `0` as a special value to disable size-based flush, so still check whether `batchSize` is larger than 0, has the special value for some backward comparability(only count-based flush), what do you think?

Reply:
Currently, use `0` as a special value to disable size-based flush, so still check whether `batchSize` is larger than 0, has the special value for some backward comparability(only count-based flush), what do you think?

Reply:
Ok, makes sense.